app/src/main/java/io/github/njackwinterofcode/robotalk/textToSpeech.java

```diff
@@ -51,43 +49,27 @@ public void onInit(int status) {
                 }
             }
         });
-
         convertButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                String data = text.getText().toString();
-                text.onEditorAction(EditorInfo.IME_ACTION_DONE);
-                Log.i("TTS", "Button Clicked" + data);
-                int speechStatus = textToSpeech.speak(data, TextToSpeech.QUEUE_FLUSH,null);
-
-               try{
-                   textToSpeech.setSpeechRate(Integer.parseInt(speed.getText().toString()));
-               }catch (Exception e){
-                   Toast.makeText(textToSpeech.this, "Speech rate taken as default.", Toast.LENGTH_SHORT).show();
-               }
-                if(speechStatus == TextToSpeech.ERROR){
-                    Log.e("TTS ", "Error in conversion");
-                }
-            }
-        });
-
-        pitch.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
-            @Override
-            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
-                textToSpeech.setPitch(progress);
-            }
-
-            @Override
-            public void onStartTrackingTouch(SeekBar seekBar) {
-
-            }
-
-            @Override
-            public void onStopTrackingTouch(SeekBar seekBar) {
-
+                speak();
             }
         });
     }
+    private void speak() {
+        String text = mtext.getText().toString();
+        float pitch = (float) mpitch.getProgress() / 50;
+        if (pitch < 0.1) {
+            pitch = 0.1f;
+        }
+        float speed = (float) mspeed.getProgress() / 50;
+        if (speed < 0.1) {
+            speed = 0.1f;
```

Comment:
What is the concept behind taking these particular values only?

Reply:
@ShivangiSingh17 Ma'am, The seekbar takes values from 0-100. I have taken 50 as my initial progress and set it to 1 using this float speed = (float) mspeed.getProgress() / 50; [Both in Pitch and Speed]. Also we don't want the inputs to be 0 because then there won't be any output. So we have checked if the Speed<0.1 then we set its value to our defined minimum i.e. 0.1f ( f for floating value).